qwerty_ast_to_mlir/src/lower/qpu.rs

```diff
@@ -480,44 +520,66 @@ fn is_fourier(basis: &Basis) -> bool {
 fn ast_basis_to_mlir(basis: &Basis) -> MlirBasis {
     let basis_elements = basis.make_explicit().canonicalize().to_vec();
 
-    let (elems, phases): (Vec<_>, Vec<_>) = basis_elements
-        .iter()
-        .map(|elem| {
-            match elem {
-                Basis::BasisLiteral { vecs, .. } if is_bell_basis(vecs) => {
-                    let std = qwerty::BuiltinBasisAttribute::new(&MLIR_CTX, qwerty::PrimitiveBasis::Bell, 2);
-                    (qwerty::BasisElemAttribute::from_std(&MLIR_CTX, std), vec![])
-                }
+    let (vec_basis_equiv_prim, prim_basis) = is_vec_basis_equiv_primitive(&basis_elements);
+    let (elems, phases) = if vec_basis_equiv_prim {
```

Comment:
This check can be done per-basis element instead of for all basis elements, if I'm making sense